src/screens/PinFlow/CurrentPinScreen.tsx

```diff
@@ -14,37 +25,95 @@ interface Props {
   appSettings: {
     isPinSet: boolean;
   };
+  setTimeCounter: (timestamp: number) => SetTimeCounterAction;
+  setFailedAttempts: (attempt: number) => SetFailedAttemptsAction;
+  timeCounter: TimeCounterState;
 }
 
 interface State {
   pin: string;
   error: string;
+  failedTimes: number;
```

Comment:
as I see you don't need local failedTimes as you already have this in redux. Reading from two sources of truth  makes it hard to debug and read the code

Reply:
failedTimes and attempts are two different thigs, failedTimes are included in one attempt. So you have attempts and attempt could have maximum three failedTimes, then the timer is opening, attempt is increasing and failedTimes are reseting.

Reply:
I will merge it due to testing process. If you have solution to make it more redable then write it there and I'm gonna apply it in other PR.